### PR TITLE
[AT359] Create reducer for instagram stories breakdown

### PR DESCRIPTION
**Pull request by gmzjuliana**
### Purpose
define default state and define action for filtering data (select metric)

### CODE REVIEW

File: packages/stories-metrics/reducer.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+const initialState = {
+  loading: true,
+  hasError: false,
+  data: {
+    totals: [],
+    daily: []
+  },
+  selectedMetrics: {
+    primary: {},
```
Comment (federicoweber) [inline packages/stories-metrics/reducer.js]: Hey @gmzjuliana, as the populated value will be a string do you think we can use `null` or empty string as default values?

File: packages/stories-metrics/reducer.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+function isMetricValid(totals, metricLabel) {
+  const metrics = totals.map(m => m.label);
+  return metrics.indexOf(metricLabel) !== -1;
+}
+
+function getSelectedMetricLabelsOnFetch(state, result) {
+  const totals = result.totals;
+  if (totals.length > 0) {
+    if (
+      state.selectedMetricLabel !== "" &&
```
Comment (federicoweber) [inline packages/stories-metrics/reducer.js]: I don't see `selectedMetricLabel` in our state 😕. Should it be `!state.selectedMetrics.primary`?

File: packages/stories-metrics/reducer.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      isMetricValid(totals, state.selectedMetricLabel)
```
Comment (federicoweber) [inline packages/stories-metrics/reducer.js]: what is the reason we need this?
Given that we are extracting the metrics from the totals, and we only support one service, my hunch is that we shouldn't really see invalid metrics.

File: packages/stories-metrics/reducer.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      return state;
+  }
+};
+
+export const actions = {
+  selectMetric(metricLabel) {
+    return {
+      type: `stories_metrics_${actionTypes.SELECT_METRIC}`,
+      metricLabel,
+      isPrimary
```
Comment (federicoweber) [inline packages/stories-metrics/reducer.js]: I think we aren't passing this down 😉

File: packages/stories-metrics/reducer.js (hunk 1 above)
Comment (gmzjuliana) [reply packages/stories-metrics/reducer.js]: yes! I used an empty string to keep what I had in `getSelectedMetricLabelsOnFetch`

File: packages/stories-metrics/reducer.js (hunk 4 above)
Comment (gmzjuliana) [reply packages/stories-metrics/reducer.js]: gotcha, thanks for catching this

**Comment by gmzjuliana** [comment]
Added these changes into #539

**Pull request closed without merging**